### PR TITLE
Expose a mobile method to fetch CafeRequest by id

### PR DESCRIPTION
**Pull request by asutula**
Really no idea what i'm doing

### CODE REVIEW

File: mobile/cafes_test.go [hunk 1]
```diff
@@ -254,6 +254,14 @@ func flushCafeRequests(limit int) (int, error) {
 		}
 	}
 
+	// make sure we can query for each
+	for g := range reqs {
+		bytes, err = cafesTestVars.mobile.CafeRequest(g)
```
Comment (carsonfarmer) [inline mobile/cafes_test.go]: since you haven't created a bytes yet, you'll need to use `bytes, err := cafesTest...`

**Comment by asutula** [comment]
Ok, I think that's a legit failure. It's not returning any bytes when I query for the CafeRequest with my new method. Not sure why, could use any ideas.

**Comment by asutula** [comment]
Could I get some feedback here @sanderpick ?

File: mobile/cafes.go [hunk 2]
```diff
@@ -147,6 +147,24 @@ func (m *Mobile) CafeRequests(limit int) ([]byte, error) {
...
+	if !m.node.Started() {
+		return nil, core.ErrStopped
+	}
+
+	cafeRequest := m.node.Datastore().CafeRequests().Get(id)
+	if cafeRequest == nil {
+		return nil, nil
+	}
+
+	bytes, err := proto.Marshal(cafeRequest)
```
Comment (sanderpick) [inline mobile/cafes.go]: You can just `return proto.Marshal(cafeRequest)`

File: mobile/cafes_test.go [hunk 3]
```diff
@@ -254,6 +255,17 @@ func flushCafeRequests(limit int) (int, error) {
 		}
 	}
 
+	// make sure we can query for each
+	for g := range reqs {
+		bytes, err := cafesTestVars.mobile.CafeRequest(g)
```
Comment (sanderpick) [inline mobile/cafes_test.go]: Here you're using a group ID where you want a single request ID.

**Comment by asutula** [comment]
In the case of an expired refresh token in the native mobile sdks, we need to ask the host application to re-register the cafe. To do that, we'd need to pass the cafe peer id to the host application. To get the cafe peer id, I need some way of getting a peer id from a cafe request id.

**Comment by asutula** [comment]
Within the context of a failed http request, the only thing I have is the cafe request id. So this would provide a way to map that to cafe information, basically

**Comment by sanderpick** [comment]
Okay, cool! We could add a method like `GetCafeFromGroup` to the datastore interface:

```
type CafeRequestStore interface {
	Queryable
	Add(req *pb.CafeRequest) error
	Get(id string) *pb.CafeRequest
	GetGroup(group string) *pb.CafeRequestList
	GetSyncGroup(group string) string
	Count(status pb.CafeRequest_Status) int
	List(offset string, limit int) *pb.CafeRequestList
	ListGroups(offset string, limit int) []string
	SyncGroupComplete(syncGroupId string) bool
	SyncGroupStatus(groupId string) *pb.CafeSyncGroupStatus
	UpdateStatus(id string, status pb.CafeRequest_Status) error
	UpdateGroupStatus(group string, status pb.CafeRequest_Status) error
	UpdateGroupProgress(group string, transferred int64, total int64) error
	AddAttempt(id string) error
	Delete(id string) error
	DeleteByGroup(groupId string) error
	DeleteBySyncGroup(syncGroupId string) error
	DeleteCompleteSyncGroups() error
	DeleteByCafe(cafeId string) error
}
```

**Comment by asutula** [comment]
Sounds good to me. This was a pretty nieve stab at something like that, so we can close this and do the proper thing.

**Comment by sanderpick** [comment]
Or just make some changes to this?

**Comment by asutula** [comment]
Ok yea sounds good. Will get back on this asap. For now, I'm updating the photos app to just refresh sessions as needed on start up (as a temporary measure)

**Comment by sanderpick** [comment]
Sounds good. If we did that plus changed the token expiration to something like 6 months we could table this for now and focus on trying new features like camera roll sync.

**Comment by asutula** [comment]
Exactly, we're on the same page.

**Comment by sanderpick** [comment]
Awesome. I'm make an issue.

**Comment by sanderpick** [comment]
Still needed @asutula ?

**Pull request closed without merging**